syntax = "proto3";

package books;

service BookService {
    rpc GetBook(BookRequest) returns (BookResponse) {}
    rpc ListBooks(Empty) returns (BookList) {}
}

message BookRequest {
    int32 id = 1;
}

message BookResponse {
    int32 id = 1;
    string title = 2;
    string author = 3;
    string publication_date = 4;
}

message BookList {
    repeated BookResponse books = 1;
}

message Empty {}
